recipe-server/normandy/health/api/views.py

```diff
@@ -15,42 +17,51 @@
 
 
 logger = logging.getLogger(__name__)
-_commit = None
-_tag = None
 
 
+@lru_cache(maxsize=1)
 def get_version_info():
-    global _commit, _tag
-    if _commit is None:
-        path = os.path.join(settings.BASE_DIR, '__version__', 'commit')
-        try:
-            with open(path) as f:
-                _commit = f.read().strip() or 'unknown'
-        except OSError:
-            _commit = 'unknown'
 
-    if _tag is None:
-        path = os.path.join(settings.BASE_DIR, '__version__', 'tag')
+    def fetch_key(name):
+        path = os.path.join(settings.BASE_DIR, '__version__', name)
+        val = None
         try:
             with open(path) as f:
-                _tag = f.read().strip() or 'unknown'
+                val = f.read().strip()
         except OSError:
-            _tag = 'unknown'
+            pass
+        return val or None
 
-    return _commit, _tag
+    try:
+        build_time = datetime.utcfromtimestamp(int(fetch_key('build_time')))
+    except (ValueError, TypeError) as e:
+        build_time = None
+
+    version_info = {
+        'commit': fetch_key('commit'),
+        'version': fetch_key('tag'),
+        'build_time': build_time,
+    }
+
+    return version_info
 
 
 @api_view(['GET'])
 def version(request):
```

Comment:
In a strong tradition of asking you to fix code you didn't touch, this could use a docstring explaining the purpose of the view.